### PR TITLE
fix(streaming): fix stream polling after split change

### LINKED ISSUE #6300: source: source reader will stall after replacing if previous reader has finished
The `SelectWithStrategy` maintains an internal state of termination for both sides. If one side is terminated, it'll never poll it in the future, for optimization.

```rust
match this.internal_state {
    InternalState::Start => {
        let next_side = (this.clos)(this.state);
        poll_inner(&mut this, next_side, cx)
    }
    InternalState::LeftFinished => match this.stream2.poll_next(cx) {
        Poll::Ready(None) => {
            *this.internal_state = InternalState::BothFinished;
            Poll::Ready(None)
        }
        a => a,
    },
    InternalState::RightFinished => match this.stream1.poll_next(cx) {
        Poll::Ready(None) => {
            *this.internal_state = InternalState::BothFinished;
            Poll::Ready(None)
        }
        a => a,
    },
    InternalState::BothFinished => Poll::Ready(None),
}
```

If the source reader has finished before split changing (or scaling), even if we replace it with a new active reader, the `SelectWithStrategy` still won't poll it anymore. So we can never get the chunks from this source! Actually, the `futures` has warned us about this.

> Note that care must be taken to avoid tampering with the state of the stream which may otherwise confuse this combinator.

https://github.com/risingwavelabs/risingwave/blob/5d8465af3fd406e3f742ba185ab176591d35f211/src/stream/src/executor/source/reader.rs#L88-L91

Since we set the event num for the Nexmark source in the deterministic scaling test, and it will finish after emitting all events, this is also why we get incorrect results if we allow scaling source in the chaos test.

- https://github.com/risingwavelabs/risingwave/pull/6062#discussion_r1006459041

https://github.com/risingwavelabs/risingwave/blob/5d8465af3fd406e3f742ba185ab176591d35f211/src/tests/simulation_scale/src/ctl_ext.rs#L101-L103

### PR DESCRIPTION
**Pull request by BugenZhao**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

As explained in #6300.

This PR also add more logs to source, enables chaos scaling test for `Source` nodes.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)
- Fix #6300

### CODE REVIEW

**Review by wangrunji0408 — APPROVED** [review]
LGTM!

File: src/connector/src/source/base.rs [hunk 1]
```diff
@@ -195,6 +195,14 @@ pub fn spawn_data_generation_stream<T: Send + 'static>(
...
     buffer_size: usize,
 ) -> impl Stream<Item = T> + Send + 'static {
+    static RUNTIME: LazyLock<Runtime> = LazyLock::new(|| {
+        tokio::runtime::Builder::new_multi_thread()
+            .thread_name("risingwave-data-generation")
+            .enable_all()
+            .build()
+            .expect("failed to build data-generation runtime")
+    });
+
```
Comment (tabVersion) [inline src/connector/src/source/base.rs:205]: so will all connectors run in a separate runtime apart from executor?

File: src/stream/src/executor/source/source_executor.rs [hunk 2]
```diff
@@ -438,6 +437,12 @@ impl<S: StateStore> SourceExecutor<S> {
     ) -> StreamExecutorResult<()> {
         if let Some(target_splits) = mapping.get(&self.ctx.id).cloned() {
             if let Some(target_state) = self.get_diff(Some(target_splits)).await? {
+                tracing::debug!(
```
Comment (tabVersion) [inline src/stream/src/executor/source/source_executor.rs:440]: suggest info! here

**Review by tabVersion — APPROVED** [review]
source part LGTM

File: src/connector/src/source/base.rs (hunk 1 above)
Comment (BugenZhao) [reply src/connector/src/source/base.rs:205]: Yes. These data generations are usually CPU-intensive and may cause scheduling problems if we put them in the same runtime as streaming actors.

File: src/stream/src/executor/source/source_executor.rs (hunk 2 above)
Comment (BugenZhao) [reply src/stream/src/executor/source/source_executor.rs:440]:
```suggestion
                tracing::info!(
```

**Pull request merged**